Create pyramid.20.06.md (#3)
KunitzAn/KunitzAn.github.io: pyramid.20.06.md

        ```
 
7. Use `@media screen` to apply a different style for a device with a screen and a window with `max-width` of 450px.

   ```CSS
   @media screen and (max-width: 450px) {
      .buttons { 
            font-size: 14px;
            width: 70px;
            height: 40px;
        }
        .input-window {
            width: 80px;
        }
        .input {
            max-width: 350px;
        }
      }
   ```

8. Apply the style to the HTML content. Open the `pyramid.html` file, click in the blank line between the `<tittle>` tag and closing `</head>` tag near the top of the page, and type:

   ```HTML
   <link rel="stylesheet" href="pyramid.css" type="text/css">
   ```
   After changing the HTML file don't forget to save it again. 

9. The code from above results in following: 

   ![styledHTMLPage](https://github.com/KunitzAn/KunitzAn.github.io/blob/main/photos-js/result.png?raw=true)
   
   Save the CSS file and open the `pyramid.html` file in your browser to check the result.

## Create and style the Pyramid
1. Create the "pyramid.js" file. Then open the "pyramid.html" file and add a link to the "pyramid.js" file before the closed `</body>` tag.  

    ```HTML

>     <script src="pyramid.js" type="text/javascript"></script>

Reviewer: More code lines

    ```

   Above the link to the JS file add the container for the pyramid.

    ```HTML
    <div class="pyramid" id="piram"></div>
    ```
    

> 2. Add an `onclick` event with a `Create` function to the `Create` button. 

Reviewer: ```suggestion
2. Handle the `onclick` event for the `Create` button and call the `create` function. 
```

                    margin: 5px;
                }
                .pyramid {
                    margin-left: 5px;
                }
            }
            ```
    
4.  Add the `create()` function in the "pyramid.js" file.

    ```JS
    function create() {
    }
    ```   
5. Let users enter the input value from 1 to 9999. If it less than 1 or more than 9999, quit the `create()` function using `return`.

    ```
    if(input.value<1 || input.value>9999){
      return;
    }
    ```
    
6. Define new variables. Use HTML DOM Document `getElementByID()` Method. Refer to the following tutorial to learn more about this method: [HTML DOM Document getElementById()](https://www.w3schools.com/jsref/met_document_getelementbyid.asp).

   * Enter the first block of code above the `create()` function: 

     ```JS
     var input = document.getElementById("inpt-wnd");
     var piram = document.getElementById("piram");
     var n = input.value;
     var createdButtons = 0 //created buttons amount
     var collection = document.getElementsByClassName('line');
     var linesCount = collection.length;
     ```
    
   * Enter the second block inside the `Create` function:

>      ```JS

Reviewer: I think it is better to display the whole code and move the descriptions in comments. It is unclear where to add all those code lines. A user don't like to think, a user wants to copy-paste and it should work :)

            ```
            
        * Delete all created text inside the `buttons` and add new button numbers.
                
            ```JS
            for(let i = 0; i<buttonCollection.length; i++) {
              buttonCollection[i].removeChild(buttonCollection[i].firstChild);
              var text = document.createTextNode(i+1);
              buttonCollection[i].appendChild(text);
            }
            ```
            
        * Change the order of the `button` elements in each line. Enter `return` to quite the function.   
        
            ```JS
              for(let i = 0; i<linesToReverse.length; i++) {
                var aline = linesToReverse[i];
                aline.style.flexDirection='row';
              }
            return; 
            ```
            
5. Check, if the `rotate` button works correctly. The first picture depicts how the pyramid looks before clicking the `Rotate` button, and the second one - after.

    ![pyraamid](https://github.com/KunitzAn/KunitzAn.github.io/blob/main/photos2/9.png?raw=true)
    ![rotated pyramid](https://github.com/KunitzAn/KunitzAn.github.io/blob/main/photos2/10.png?raw=true)

> ## Let users remove the pyramid by clicking the "Remove all" button

Reviewer: ## Remove the pyramid

    * Use `if` statement to check, if the `input` value is changed. 
        
        ```JS
        if(!input.value==0) {
        }
        ```
        
    * Write the code that should be executed, if the condition is true:
        * Choose the element with the `piram` id.
        * Define the last button in the `piram` element using the `Eleent.lastElementChild` property.
        *  Sequentially delete every pyramid last child until the pyramid is empty.
        
            ```JS
            if(!input.value==0) {
                var piramid = document.getElementById('piram');
                var child = piram.lastElementChild;
                while (child) {
                    piramid.removeChild(child);
                    child = piramid.lastElementChild;
                }
            }
            ```
    
5. Open the app in your browser and check, if all the buttons work correctly. If someting goes wrong, compare your HTML, CSS and JS files with the files in the repository: ["Button pyramid"](https://github.com/KunitzAn/KunitzAn.github.io/tree/main/Button%20Pyramid).

## Deploy the App to GitHub pages

> 1. If you don't have a GitHub repository, refer to this instruction to learn how to create it: [Creating a new repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-new-repository).

Reviewer: ```suggestion
1. You need a GitHub repository to deploy your application on GitHub Pages. If you don't have a GitHub repository, refer to this instruction to learn how to create it: [Creating a new repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-new-repository).
```

> 2. When you create a repository on GitHub.com, it exists as a remote repository. Refer to the following tutorial to learn how to clone your repository to create a local copy on your computer: [Cloning a repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository).

Reviewer: ```suggestion
2. [Clone the newly created repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository).
```

> 3. Then add your project folder to the local copy of your repository:

Reviewer: ```suggestion
3. Add your project folder to the local copy of your repository:
```

   * Paste here the project folder. 

    ![fing local repo](https://github.com/KunitzAn/KunitzAn.github.io/blob/main/photos2/4%20%E2%80%94%20%D0%BA%D0%BE%D0%BF%D0%B8%D1%8F.png?raw=true)
    
4. Open your GitHub repository folder in VS Code and choose the project folder.

    ![open git repo 1](https://github.com/KunitzAn/KunitzAn.github.io/blob/main/photos2/5%20%E2%80%94%20%D0%BA%D0%BE%D0%BF%D0%B8%D1%8F.png?raw=true)
    
    ![open git repo 2](https://github.com/KunitzAn/KunitzAn.github.io/blob/main/photos2/6%20%E2%80%94%20%D0%BA%D0%BE%D0%BF%D0%B8%D1%8F.png?raw=true)
    
5. Click on `Source control` [1], then click `+` [2] and `Commit` [3] button to stage all changes. 

    ![commit changes](https://github.com/KunitzAn/KunitzAn.github.io/blob/main/photos2/7%20%E2%80%94%20%D0%BA%D0%BE%D0%BF%D0%B8%D1%8F.png?raw=true)
    
6. Enter the commit massage [1] and click `Sync Changes` [2] to commit all changes.

    ![sync changes](https://github.com/KunitzAn/KunitzAn.github.io/blob/main/photos2/8%20%E2%80%94%20%D0%BA%D0%BE%D0%BF%D0%B8%D1%8F.png?raw=true)

7. Open your repository on GitHub in your browser to access the project folder.

>     ![Project folder on GitHub](https://github.com/KunitzAn/KunitzAn.github.io/blob/main/photos2/11%20%E2%80%94%20%D0%BA%D0%BE%D0%BF%D0%B8%D1%8F.png?raw=true)

Reviewer: I think it is not deploy. It is how to add your project to GitHub. To deploy on GitHub Pages, you have to do some additional steps. You need to rename the section or add missing steps.
